pinot-core/src/main/java/org/apache/pinot/core/transport/InstanceRequestHandler.java

```diff
@@ -54,19 +65,24 @@
  * The {@code InstanceRequestHandler} is the Netty inbound handler on Pinot Server side to handle the serialized
  * instance requests sent from Pinot Broker.
  */
+@ChannelHandler.Sharable
 public class InstanceRequestHandler extends SimpleChannelInboundHandler<ByteBuf> {
   private static final Logger LOGGER = LoggerFactory.getLogger(InstanceRequestHandler.class);
 
   // TODO: make it configurable
   private static final int SLOW_QUERY_LATENCY_THRESHOLD_MS = 100;
 
+  private final String _instanceName;
   private final TDeserializer _deserializer;
   private final QueryScheduler _queryScheduler;
   private final ServerMetrics _serverMetrics;
   private final AccessControl _accessControl;
+  private final boolean _enableQueryCancellation;
+  private final Map<String, Future<byte[]>> _queryFuturesById = new ConcurrentHashMap<>();
```

Comment:
When query cancellation is not enabled, let's set it to `null` to avoid the unused garbage.
(Optional) We may also use whether it is `null` to identify whether query cancellation is enabled, this should have slightly better performance

Reply:
sgtm